diesel/src/query_builder/debug_query.rs

```diff
@@ -34,12 +61,15 @@ where
     T: QueryFragment<DB>,
 {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        let mut query_builder = DB::QueryBuilder::default();
-        let backend = DB::default();
-        QueryFragment::<DB>::to_sql(self.query, &mut query_builder, &backend)
-            .map_err(|_| fmt::Error)?;
-        let debug_binds = DebugBinds::<_, DB>::new(self.query);
-        write!(f, "{} -- binds: {:?}", query_builder.finish(), debug_binds)
+        // This is not using the `?` operator to reduce the code size of this
+        // function, which is getting copies a lot due to monomorphization.
+        let query = match self.query() {
+            Ok(query) => query,
+            Err(err) => return Err(err),
+        };
```

Comment:
I tried to use a `tri!()` macro instead (see https://github.com/serde-rs/serde/blob/v1.0.217/serde/src/lib.rs#L290-L301), but that slightly bumped the numbers upwards again, so I'm not sure whether it's worth it for the readability benefit.